### PR TITLE
Nested table max-width prop

### LINKED ISSUE #3129: [Table] Nested table test wrap
### What package is this for?

- [x] React
- [ ] Angular

### Describe the bug

For table nested table, at some point column can get overridden by one with bigger content
### To Reproduce

1. Go to StatefulTable with row nesting and resizable columns with initial width
2. Click on expand to one with 4 levels
3. See the error

### Expected behavior

Max-width of the span should be based on parent width - current offset

### Environment/versions:

- OS:  MacOS
- Browser: chrome
- `carbon-addons-iot-react` version:  `v2.147.0-next.79`

### Additional context

image

### Specific timeline issues / requests

Do you want this work within a specific time period? Yes.
Is it related to an upcoming release? MAS Monitor.

### PR DESCRIPTION
**Pull request by cgirani**
Closes #3129

**Summary**

- Add a max-width in the span, based on the nested offset.

Before:
image

After:
image

**Change List (commits, features, bugs, etc)**

- fix(tablerow): max-width for nested table

**Acceptance Test (how to verify the PR)**

- Go to StatefulTable - with row nesting and resizable columns with initial width
- Open an item with at least 4 nested levels and see the width is not overwritten the next column

**Regression Test (how to make sure this PR doesn't break old functionality)**

- Make sure the nested table keeps working properly.

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

**Review by kevinsperrine — APPROVED** [review]
Thanks for the help. Looks great!

**Review by bjornalm — COMMENTED** [review]
Fix looks clean and simple. I just wanna test it properly before approving. I've restarted the failed deployment.

**Review by bjornalm — APPROVED** [review]

**Review by davidicus — APPROVED** [review]

File: packages/react/src/components/Table/TableBody/TableBodyRow/TableBodyRow.jsx [hunk 1]
```diff
@@ -381,6 +381,7 @@ const StyledExpansionTableRow = styled(({ hasRowSelection, ...props }) => <Table
 const StyledNestedSpan = styled.span`
   position: relative;
   left: ${(props) => props.nestingOffset}px;
+  max-width: calc(100% - ${(props) => props.nestingOffset}px);
```
Comment (davidicus) [inline packages/react/src/components/Table/TableBody/TableBodyRow/TableBodyRow.jsx:384]: We should not be doing this with styled components. This prop should be added as a css custom property.

**Review by davidicus — CHANGES_REQUESTED** [review]
We do not support styled components and these will be removed in the future. Any new features that need a prop value should be done using css custom properties

**Pull request merged**